Fix clipped box shadows in ContentCardGroupCarousel

jonboiser opened the pull request:
### Summary

This slightly changes some styles in ContentCardGroupCarousel so that the box shadows are not cut off at the top and left-most edges (right-most for RTL).

**Before**

<img width="651" alt="screen shot 2018-10-01 at 12 13 09 pm" src="https://user-images.githubusercontent.com/10248067/46310369-fbba4680-c573-11e8-96c1-c749f877bb6b.png">

**After (with more opacity)**
<img width="586" alt="screen shot 2018-10-03 at 10 47 43 am" src="https://user-images.githubusercontent.com/10248067/46428806-cb9cb000-c6f9-11e8-8fa1-00d67a198927.png">


### Reviewer guidance

1. Check it on RTL
1. Check it on different numbers of cards visible

### References

Fixes #3918 
…

----

### Contributor Checklist

- [x] Contributor has fully tested the PR manually
- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [X] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [X] If there are any front-end changes, before/after screenshots are included
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

jonboiser committed:
Fix clipped box shadows in ContentCardGroupCarousel

indirectlylit commented:
This is awesome. However, looks like the shadows are still very slightly clipped on the side and top?

jonboiser commented:
yeah, i can see it if remove the transparency from the shadows. made need to add about 3px more (8px total to fit the shadows)

indirectlylit commented:
@DXCanas would you mind reviewing this?

DXCanas reviewed (commented):
Shadows look good, but next/previous buttons aren't aligned on either side now. IMO, that's more distracting than the shadows.
![image](https://user-images.githubusercontent.com/9877852/46441000-08789f00-c71a-11e8-81b0-f8b99856f264.png)

If @khangmach's got no issue, i'll approve and merge.

khangmach commented:
@DXCanas is it possible to get the right arrow back to where it was and still fix the clipping? That's buggin' me too. 

On a side note, is this extra white space just part of the thumbnail?

<img width="111" alt="screen shot 2018-10-03 at 3 06 24 pm" src="https://user-images.githubusercontent.com/25045166/46442294-00226300-c71e-11e8-80ea-527a8de83a7c.png">

DXCanas committed:
Resolves #3918
Using padding and negative margin to give shadows room to breathe.
Revert "Fix clipped box shadows in ContentCardGroupCarousel"

This reverts commit 2299ce7cd612e3162da7b1f717bd8c0b53e479c4.

DXCanas commented:
@jonboiser @khangmach @indirectlylit 
Commited an alternative fix. Feel free to kill it if it you prefer JB's; it embraces a CSS quirk. Has the downside of clipping the controls if the parent applies a margin, padding, or overflow.
![alt fix](https://user-images.githubusercontent.com/9877852/46443589-0f57df80-c723-11e8-8f57-ce2d0ff7afb7.png)

indirectlylit commented:
>  it embraces a CSS quirk

can you elaborate?

DXCanas commented:
> can you elaborate?

Just that it uses negative margin to offset the padding, essentially to extend the boundaries used in `overflow: hidden` far enough out as to not clip the controls

indirectlylit commented:
sounds fine to me! Same trick I used in the grid :)

https://github.com/learningequality/kolibri/blob/develop/kolibri/core/assets/src/views/KGrid/index.vue#L94-L98
